### PR TITLE
GeometricDetExtra, PGeometricDetExtra schema removal

### PR DESCRIPTION
**Pull request by vargasa**
#### PR description:

Addresses https://github.com/cms-sw/cmssw/issues/31567 

FYI: @ianna, @ghugo83, @cvuosalo

### CODE REVIEW

**Comment by ianna** [comment]
please check

**Comment by ianna** [comment]
please test

File: Geometry/TrackerGeometryBuilder/test/ModuleInfo.cc
Comment (mmusich) [inline Geometry/TrackerGeometryBuilder/test/ModuleInfo.cc:168]: isn't this broken by #31573?
Now it's `geographicalId()`

File: SLHCUpgradeSimulations/Geometry/test/ModuleInfo_Phase2.cc
Comment (mmusich) [inline SLHCUpgradeSimulations/Geometry/test/ModuleInfo_Phase2.cc:276]: isn't this broken by #31573?
Now it's `geographicalId()`

**Comment by mmusich** [comment]
-1
@vargasa, if the Global Tags are not cleaned by the record first, I suspect this will fail... 

BTW: why have you proceeded without explicit aknowledgement from the DPG?

**Comment by mmusich** [comment]
@vargasa please remove from the history my commit 02601ad already merged.

**Comment by mmusich** [comment]
@christopheralanwest,@yuanchao, @tocheng, @ggovi
This PR should be put on hold, until all the Global Tags in autoCond have been cleaned by the removed record

**Comment by vargasa** [comment]
> BTW: why have you proceeded without explicit aknowledgement from the DPG?

Hi @mmusich, no answer was collected from the announcement https://hypernews.cern.ch/HyperNews/CMS/get/tracker-performance/2388.html except your concern in a private conversation (Sep 29, 2020 email) regarding its usage which I tried to address to the best of my knowledge. Am I missing something? or Do we need to go through an approval process for this PR? If so, I would still suggest to run the tests so both things can be done in parallel. If you don't agree I can also close the PR but that may put additional waiting time within our already tight timeline

**Comment by mmusich** [comment]
>  Am I missing something? or Do we need to go through an approval process for this PR? If so, I would still suggest to run the tests so both things can be done in parallel. 

@vargasa, this PR "as-is" is 100% going to fail. We need to clean all the records in DB. If you'd have alerted us in time we could have asked AlCA to clean the autoCond Global Tag *before* submitting the PR....

**Comment by vargasa** [comment]
@mmusich the submission of the PR doesn't represent a problem by itself. We didn't have that information prior to this. It's now a draft, please let me know if there is any other requirement to proceed with this

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by christopheralanwest** [comment]
> > BTW: why have you proceeded without explicit aknowledgement from the DPG?
> 
> Hi @mmusich, no answer was collected from the announcement https://hypernews.cern.ch/HyperNews/CMS/get/tracker-performance/2388.html except your concern in a private conversation (Sep 29, 2020 email) regarding its usage which I tried to address to the best of my knowledge. Am I missing something? or Do we need to go through an approval process for this PR? If so, I would still suggest to run the tests so both things can be done in parallel. If you don't agree I can also close the PR but that may put additional waiting time within our already tight timeline

AlCa will need to remove the relevant records from all 38 different GTs in autoCond. It is easiest to make such a change when there are no open PRs that would conflict with this change, which is not currently the case. I don't understand the reason for urgency mentioned above--could you explain? On what time scale are you hoping to have this removal accomplished?

**Comment by ianna** [comment]
> > > BTW: why have you proceeded without explicit aknowledgement from the DPG?
> > 
> > 
> > Hi @mmusich, no answer was collected from the announcement https://hypernews.cern.ch/HyperNews/CMS/get/tracker-performance/2388.html except your concern in a private conversation (Sep 29, 2020 email) regarding its usage which I tried to address to the best of my knowledge. Am I missing something? or Do we need to go through an approval process for this PR? If so, I would still suggest to run the tests so both things can be done in parallel. If you don't agree I can also close the PR but that may put additional waiting time within our already tight timeline
> 
> AlCa will need to remove the relevant records from all 38 different GTs in autoCond. It is easiest to make such a change when there are no open PRs that would conflict with this change, which is not currently the case. I don't understand the reason for urgency mentioned above--could you explain? On what time scale are you hoping to have this removal accomplished?

This is directly related to DD4HEP migration. The time scale is defined by 11_2. We have limited resources and would like to move fast. I have contacted @ggovi for an advice how to proceed. It seems that opening a PR is the fastest option to collect feedback 😀

Please advise which part of this PR must stay to avoid pressure on the GT update. Thanks!

**Comment by civanch** [comment]
@vargasa , @ianna , is it possible to make an intermediate step, when GeometricDetExtra is not used but not removed from GT?

**Comment by ianna** [comment]
> @vargasa , @ianna , is it possible to make an intermediate step, when GeometricDetExtra is not used but not removed from GT?

Yes. However, in this case we need to keep the minimum required. @ggovi and @christopheralanwest please, mark the files in this PR that must stay in CMSSW. Thanks!

**Comment by christopheralanwest** [comment]
> > @vargasa , @ianna , is it possible to make an intermediate step, when GeometricDetExtra is not used but not removed from GT?
> 
> Yes. However, in this case we need to keep the minimum required. @ggovi and @christopheralanwest please, mark the files in this PR that must stay in CMSSW. Thanks!

I've prepared the necessary queues and will submit the PR to update `autoCond` once an IB with PR #31645 is available.

A potential complication is that sometimes HLT requires running on GTs from old releases. Without a definition of `PGeometricDetExtra` in CMSSW, GTs that contain that record will no longer be readable, even if the record is not actually used. @cms-sw/hlt-l2 Do you have any concerns with removing `PGeometricDetExtra` entirely or does its definition need to be retained? As a test, I have successfully run `addOnTests.py` and `runTheMatrix.py` after removing the record from these GTs: https://github.com/cms-sw/cmssw/blob/2037e05d590a623f1024c5228ca9fbcff0de7ab9/Configuration/AlCa/python/autoCond.py#L39-L46

**Comment by cvuosalo** [comment]
What is the risk in deleting these records? We are trying to complete the Run 3 DD4hep migration in a very short time frame, and we should not take unnecessary risks. If the records are deleted and then the missing records cause problems, reverting and fixing the problems will take time and effort that we cannot afford.

**Comment by christopheralanwest** [comment]
Removing the records from the GTs in master poses essentially no risk because they are unused--if they were not, the matrix workflows would crash. PR #31780 removes these records from all GTs in autoCond. A problem will arise if an old GT is used in a release built after this PR is merged, but to resolve this issue AlCa could simply prepare a new version of the old GT that excludes the obsolete record.

**Comment by mmusich** [comment]
test parameters:
* pull_request = #31780

**Comment by mmusich** [comment]
@cmsbuild, please test

File: Geometry/TrackerGeometryBuilder/test/ModuleInfo.cc
Comment (mmusich) [inline Geometry/TrackerGeometryBuilder/test/ModuleInfo.cc:206]: `density_units` is now unused.
Please remove from line 80 of this file: https://github.com/cms-sw/cmssw/pull/31729/files#diff-35e53a427c13b6405f6885a94301343db11c1dae1173319765299c82ac39c5aeR80

File: SLHCUpgradeSimulations/Geometry/test/ModuleInfo_Phase2.cc
Comment (mmusich) [inline SLHCUpgradeSimulations/Geometry/test/ModuleInfo_Phase2.cc:327]: `density_units` is now unused, please remove it from L90 of this file:
https://github.com/cms-sw/cmssw/pull/31729/files#diff-6b0322842aa80408bfebe8127dd0ad0467828812b3498aafae6251228e33a137R90

This triggers the following errors in compilation: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-abcf69/9934/clang-new-warnings.log

**Review by mmusich — CHANGES_REQUESTED** [review]
@vargasa can you please provide a *compiling* branch?
How to reproduce the issue:
```
cmsrel CMSSW_11_2_X_2020-10-13-2300
cd CMSSW_11_2_X_2020-10-13-2300/src/
cmsenv
git cms-merge-topic 31729
scramv1 b -j 20
...
```
leads to:
```
In file included from /tmp/musich/CMSSW_11_2_X_2020-10-13-2300/src/SLHCUpgradeSimulations/Geometry/test/ModuleInfo_Phase2.cc:45:
/tmp/musich/CMSSW_11_2_X_2020-10-13-2300/poison/Geometry/TrackerNumberingBuilder/interface/CmsTrackerDebugNavigator.h:1:2: error: #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
 #error THIS FILE HAS BEEN REMOVED FROM THE PACKAGE.
  ^~~~~
```

in addition, can you please fix the couple of clang warnings below?

**Comment by mmusich** [comment]
@vargasa this still fails code-checks, can you apply the patch?

**Comment by vargasa** [comment]
@mmusich please wait. It is still building locally to make sure it doesn't happen the same. For some reason I was not seeing these compilation errors before. I'll let you know when it is ready :+1:

**Comment by vargasa** [comment]
@mmusich ready. I double checked, it compiles locally with no warnings. Please ping me again if you see any problem

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by christopheralanwest** [comment]
For the `UnitTestConcurrentIOVsCondCore` unit test, you will need to modify https://github.com/cms-sw/cmssw/blob/526b69270db14b611d10847a81d78821fb36d49e/CondCore/ESSources/test/TestConcurrentIOVsCondCore_cfg.py#L69 to point to the GT `112X_dataRun2_v5`. I think that we want to use the actual GT name rather than the key names in `autoCond` because in principle the content of `112X_dataRun2_v5` referenced in the unit test could change.

I will let @mmusich comment on the other unit test failures.

**Comment by vargasa** [comment]
@christopheralanwest thanks. Just did it. Now that one runs locally. There are a couple throwing:
```
----- Begin Fatal Exception 14-Oct-2020 20:36:07 CEST-----------------------
An exception of category 'PluginLibraryLoadError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing ESSource: class=PoolDBESSource label='GlobalTag'
   [2] While attempting to load plugin PGeometricDetExtraRcd@NewProxy
Exception Message:
unable to load /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-14-1100/lib/slc7_amd64_gcc820/poisoned/plugin-poisoned-CondCoreGeometryPluginsPlugins.so because /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-14-1100/lib/slc7_amd64_gcc820/poisoned/plugin-poisoned-CondCoreGeometryPluginsPlugins.so: cannot open shared object file: No such file or directory
```
However, all references to `PGeometricDetExtraRcd` were removed. It is the same error that `UnitTestConcurrentIOVsCondCore` was throwing before

**Comment by christopheralanwest** [comment]
I think that error is due to loading an old global tag at:

https://github.com/cms-sw/cmssw/blob/18f41ede8ac62f7097030abe8f052451c78e08ad/CalibTracker/SiStripChannelGain/test/testSSTGain_PCL_FromRECO.sh#L7

The global tag should be changed to `111X_dataRun3_Express_v4`.

**Comment by mmusich** [comment]
@vargasa @christopheralanwest 
regarding the failures:

> ---> test UnitTestConcurrentIOVsCondCore had ERRORS

this, as pointed out in https://github.com/cms-sw/cmssw/pull/31729#issuecomment-708664358 can be solved by using the new run2 data GT: `112X_dataRun2_v5`, but I do not quite understand what is the advantage of having an actual GT name rather than the key names in `autoCond`. As just demonstrated in this PR, hardcoded GTs do not age well. 
Actually it seems to me, that the entire concept of using a GT for this test is an overkill, as what the original author actually wants to access is just a multi-IOV per-LS record (in this case the `BeamSpotRcd`).
I have followed up with a question for the author in the orignial PR here: https://github.com/cms-sw/cmssw/pull/28010/files#r505439543

> ---> test GeometryTrackerGeometryBuilderTestDriver had ERRORS

the error here is: 
```
----- Begin Fatal Exception 14-Oct-2020 20:36:49 CEST-----------------------
An exception of category 'EventSetupConflict' occurred while
   [0] Calling beginJob
Exception Message:
two EventSetup Producers want to deliver type="GeometricDet" label=""
 from record IdealGeometryRecord. The two providers are 
1) type="TrackerGeometricDetESModule" label=""
2) type="TrackerGeometricDetESModule" label="trackerNumberingGeometryDB"
Please either
   remove one of these Producers
   or find a way of configuring one of them so it does not deliver this data
   or use an es_prefer statement in the configuration to choose one.
----- End Fatal Exception -------------------------------------------------
```

you can introduce (as suggested by the framework error) an `ESPrefer` for the label you wish to retain: adding this line:
```python
process.es_prefer_geomdet = cms.ESPrefer("TrackerGeometricDetESModule","")
``` 
here: https://github.com/cms-sw/cmssw/blob/0cc0d777a0b181a828c9750130fe88a91690b5cc/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDB_cfg.py#L15-L18

> ---> test testSSTGainPCL_fromRECO had ERRORS
> ---> test checkMultiRunHarvestingOutput had ERRORS

as for this you can update the GT name to be `111X_dataRun3_Express_v4` as pointed out in https://github.com/cms-sw/cmssw/pull/31729#issuecomment-708677819, though in the same spirit of above I'd rather ask you to change it into an `autoCond` key:
```diff
iff --git a/CalibTracker/SiStripChannelGain/test/testSSTGain_PCL_FromRECO.sh b/CalibTracker/SiStripChannelGain/test/testSSTGain_PCL_FromRECO.sh
index 690462fca00..9fa8fb8b17b 100755
--- a/CalibTracker/SiStripChannelGain/test/testSSTGain_PCL_FromRECO.sh
+++ b/CalibTracker/SiStripChannelGain/test/testSSTGain_PCL_FromRECO.sh
@@ -4,4 +4,4 @@ function die { echo $1: status $2 ;  exit $2; }
 (cmsRun ${LOCAL_TEST_DIR}/testSSTGain_PCL_FromRECO_cfg.py era=A) || die 'Failure running cmsRun testSSTGain_PCL_FromRECO_cfg.py era=A' $?
 (cmsRun ${LOCAL_TEST_DIR}/testSSTGain_PCL_FromRECO_cfg.py era=B) || die 'Failure running cmsRun testSSTGain_PCL_FromRECO_cfg.py era=B' $?
 (cmsRun ${LOCAL_TEST_DIR}/testSSTGain_HARVEST_FromRECO.py 0) || die 'Failure running cmsRun testSSTGain_HARVEST_FromRECO.py 0' $?
-(cmsRun ${LOCAL_TEST_DIR}/testSSTGain_MultiRun_ALCAHARVEST.py globalTag=102X_dataRun2_Express_v3) || die 'Failure running cmsRun testSSTGain_MultiRun_ALCAHARVEST.py 0' $?
+(cmsRun ${LOCAL_TEST_DIR}/testSSTGain_MultiRun_ALCAHARVEST.py globalTag=auto:run3_data_express) || die 'Failure running cmsRun testSSTGain_MultiRun_ALCAHARVEST.py 0' $?
```

**Comment by wddgit** [comment]
I submitted a fix for UnitTestConcurrentIOVsCondCore, #31806

**Comment by vargasa** [comment]
> though in the same spirit of above I'd rather ask you to change it into an `autoCond` key

Hi @mmusich. Thanks. Can you point me to an example on how to do that?

**Comment by mmusich** [comment]
@vargasa 

> Can you point me to an example on how to do that?

I provided you the exact git patch to apply at the end of https://github.com/cms-sw/cmssw/pull/31729#issuecomment-709137245

**Comment by vargasa** [comment]
> I provided you the exact git patch to apply at the end of [#31729 (comment)](https://github.com/cms-sw/cmssw/pull/31729#issuecomment-709137245)

It didn't work tho:

```
===== Test "testSSTGainPCL_fromRECO" ====
argument 0: testSSTGainPCL_fromRECO
argument 1: /bin/bash
argument 2: CalibTracker/SiStripChannelGain/test
argument 3: testSSTGain_PCL_FromRECO.sh
shell is: /bin/bash
Current directory is: /uscms_data/d3/nsf/ac8qnc90c/CMSSW_11_2_X_2020-10-09-1100
topdir is: /uscms_data/d3/nsf/ac8qnc90c/CMSSW_11_2_X_2020-10-09-1100
testdir is: /uscms_data/d3/nsf/ac8qnc90c/CMSSW_11_2_X_2020-10-09-1100/src/CalibTracker/SiStripChannelGain/test
tmpdir is: /uscms_data/d3/nsf/ac8qnc90c/CMSSW_11_2_X_2020-10-09-1100/tmp/slc7_amd64_gcc820
testbin is: /uscms_data/d3/nsf/ac8qnc90c/CMSSW_11_2_X_2020-10-09-1100/test/slc7_amd64_gcc820
Running script: /uscms_data/d3/nsf/ac8qnc90c/CMSSW_11_2_X_2020-10-09-1100/src/CalibTracker/SiStripChannelGain/test/testSSTGain_PCL_FromRECO.sh
----- Begin Fatal Exception 15-Oct-2020 11:47:06 CDT-----------------------
An exception of category 'PluginLibraryLoadError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing ESSource: class=PoolDBESSource label='GlobalTag'
   [2] While attempting to load plugin PGeometricDetExtraRcd@NewProxy
Exception Message:
unable to load /uscms_data/d3/nsf/ac8qnc90c/CMSSW_11_2_X_2020-10-09-1100/lib/slc7_amd64_gcc820/poisoned/plugin-poisoned-CondCoreGeometryPluginsPlugins.so because /uscms_data/d3/nsf/ac8qnc90c/CMSSW_11_2_X_2020-10-09-1100/lib/slc7_amd64_gcc820/poisoned/plugin-poisoned-CondCoreGeometryPluginsPlugins.so: cannot open shared object file: No such file or directory
----- End Fatal Exception -------------------------------------------------
Failure running cmsRun testSSTGain_HARVEST_FromRECO.py 0: status 65
status = 16640

---> test testSSTGainPCL_fromRECO had ERRORS
```
Same message is thrown now for `GeometryTrackerGeometryBuilderTestDriver`

**Comment by mmusich** [comment]
@vargasa

> It didn't work tho:

 don't foget to merge locally this commit 32b91acd007f51c4688596f26058f76ae24615bc from https://github.com/cms-sw/cmssw/pull/31780!!

**Comment by mmusich** [comment]
@vargasa, why don't just push the mods here? The bot will take care of merging #31780 in the PR tests

File: CondCore/ESSources/test/TestConcurrentIOVsCondCore_cfg.py
Comment (vargasa) [inline CondCore/ESSources/test/TestConcurrentIOVsCondCore_cfg.py:69]: I'll drop this one. So it doesn't conflict with @wddgit PR

**Comment by vargasa** [comment]
>@mmusich why don't just push the mods here?

Done

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by vargasa** [comment]
@mmusich Can we test it with https://github.com/cms-sw/cmssw/pull/31806 too?

**Comment by mmusich** [comment]
> Can we test it with #31806 too?

Technically yes, though I have seen funny merges when more than one PR is tested.  There should be only one failing test remaining in this round.  I suppose we can retest after #31806 is merged.

**Comment by christopheralanwest** [comment]
> I found errors in the following unit tests:
> 
> ---> test UnitTestConcurrentIOVsCondCore had ERRORS
> ---> test GeometryTrackerGeometryBuilderTestDriver had ERRORS

Previously the unit test error logs showed

```
===== Test "cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-14-1100/src/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDB_cfg.py " ====
----- Begin Fatal Exception 14-Oct-2020 20:36:49 CEST-----------------------
An exception of category 'EventSetupConflict' occurred while
   [0] Calling beginJob
Exception Message:
two EventSetup Producers want to deliver type="GeometricDet" label=""
 from record IdealGeometryRecord. The two providers are 
1) type="TrackerGeometricDetESModule" label=""
2) type="TrackerGeometricDetESModule" label="trackerNumberingGeometryDB"
Please either
   remove one of these Producers
   or find a way of configuring one of them so it does not deliver this data
   or use an es_prefer statement in the configuration to choose one.
----- End Fatal Exception -------------------------------------------------
>>> processed 0 events
Failure using cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-14-1100/src/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDB_cfg.py: status 65
status = 16640

---> test GeometryTrackerGeometryBuilderTestDriver had ERRORS
```

but now the test `Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDB_cfg.py` completes successfully and the error is in the next test in the same package:

```
===== Test "cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-15-1100/src/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDDD_cfg.py " ====
/cvmfs/cms-ib.cern.ch/week0/slc7_amd64_gcc820/cms/cmssw-patch/CMSSW_11_2_X_2020-10-15-1100/python/Configuration/Geometry/GeometryExtended_cff.py:2: UserWarning: ATTENTION: This job uses Run 1 Geometry.
  warnings.warn('ATTENTION: This job uses Run 1 Geometry.')
----- Begin Fatal Exception 16-Oct-2020 00:20:31 CEST-----------------------
An exception of category 'EventSetupConflict' occurred while
   [0] Calling beginJob
Exception Message:
two EventSetup Producers want to deliver type="GeometricDet" label=""
 from record IdealGeometryRecord. The two providers are 
1) type="TrackerGeometricDetESModule" label=""
2) type="TrackerGeometricDetESModule" label="trackerNumberingGeometry"
Please either
   remove one of these Producers
   or find a way of configuring one of them so it does not deliver this data
   or use an es_prefer statement in the configuration to choose one.
----- End Fatal Exception -------------------------------------------------
----- Begin Fatal Exception 16-Oct-2020 00:20:31 CEST-----------------------
An exception of category 'EventSetupConflict' occurred while
   [0] Calling beginJob
Exception Message:
two EventSetup Producers want to deliver type="GeometricDet" label=""
 from record IdealGeometryRecord. The two providers are 
1) type="TrackerGeometricDetESModule" label=""
2) type="TrackerGeometricDetESModule" label="trackerNumberingGeometry"
Please either
   remove one of these Producers
   or find a way of configuring one of them so it does not deliver this data
   or use an es_prefer statement in the configuration to choose one.
----- End Fatal Exception -------------------------------------------------
>>> processed 0 events
>>> processed 0 events
Failure using cmsRun /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_2_X_2020-10-15-1100/src/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDDD_cfg.py: status 65
status = 16640

---> test GeometryTrackerGeometryBuilderTestDriver had ERRORS
```

After correcting `testTrackerModuleInfoDDD_cfg.py`, please verify in local tests that there are not any additional similar failures prior to pushing your updates.

**Comment by mmusich** [comment]
@christopheralanwest 

> After correcting `testTrackerModuleInfoDDD_cfg.py`, please verify in local tests that there are not any additional similar failures prior to pushing your updates.

in order to help @vargasa with the PR, I had a look, by locally merging our commit 32b91ac from #31780 on top of this branch and re-running the unit tests by hand (as this checks out the entire observable universe, in order to speed-up tests, I launch them by hand via
```csh
foreach i ( `ls ./Geometry/TrackerGeometryBuilder/test | grep test` )
    cmsRun $i >& $i.log&
 end
```
): 

The resulting log files are available at `/afs/cern.ch/user/m/musich/public/forAndres/`.
The tests which fail are:
```
$ grep Fatal * | awk -F" " '{print $1}' | sort -n | uniq -c
      4 testTrackerModuleInfoDDD_cfg.py.log:-----
      2 testTrackerModuleInfoFromDDDNoPrintDDD_cfg.py.log:-----

```

This should be resolved by applying the following patch:
```diff 
diff --git a/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDDD_cfg.py b/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDDD_cfg.py
index cf146b437ce..320c6fa075d 100644
--- a/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDDD_cfg.py
+++ b/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoDDD_cfg.py
@@ -14,6 +14,8 @@ process.TrackerGeometricDetESModule = cms.ESProducer( "TrackerGeometricDetESModu
                                                       fromDDD = cms.bool( True )
                                                      )
 
+process.es_prefer_geomdet = cms.ESPrefer("TrackerGeometricDetESModule","")
+
 process.load("Alignment.CommonAlignmentProducer.FakeAlignmentSource_cfi")
 process.preferFakeAlign = cms.ESPrefer("FakeAlignmentSource") 
 
diff --git a/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoFromDDDNoPrintDDD_cfg.py b/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoFromDDDNoPrintDDD_cfg.py
index 5276ccd84aa..74d71bd5195 100644
--- a/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoFromDDDNoPrintDDD_cfg.py
+++ b/Geometry/TrackerGeometryBuilder/test/python/testTrackerModuleInfoFromDDDNoPrintDDD_cfg.py
@@ -10,6 +10,7 @@ process.source = cms.Source("EmptySource")
 process.TrackerGeometricDetESModule = cms.ESProducer( "TrackerGeometricDetESModule",
                                                       fromDDD = cms.bool( True )
                                                      )
+process.es_prefer_geomdet = cms.ESPrefer("TrackerGeometricDetESModule","")
 
 process.load("Alignment.CommonAlignmentProducer.FakeAlignmentSource_cfi")
 process.preferFakeAlign = cms.ESPrefer("FakeAlignmentSource") 

```

**Comment by vargasa** [comment]
Hi @mmusich. Great! Thanks! I'll apply the patch, rerun the tests locally and update the PR

**Comment by vargasa** [comment]
When running locally `cmsRun testTrackerModuleInfoDB_cfg.py` I get 

```                                                                                                                      
----- Begin Fatal Exception 16-Oct-2020 08:38:53 CDT-----------------------
An exception of category 'PluginLibraryLoadError' occurred while
   [0] Constructing the EventProcessor
   [1] Constructing ESSource: class=PoolDBESSource label='GlobalTag'
   [2] While attempting to load plugin PGeometricDetExtraRcd@NewProxy
Exception Message:
unable to load /uscms_data/d3/nsf/grt82kjasoa/CMSSW_11_2_X_2020-10-09-1100/lib/slc7_amd64_gcc820/poisoned/plugin-poisoned-CondCoreGeometryPluginsPlugins.so because /uscms_data/d3/
nsf/grt82kjasoa/CMSSW_11_2_X_2020-10-09-1100/lib/slc7_amd64_gcc820/poisoned/plugin-poisoned-CondCoreGeometryPluginsPlugins.so: cannot open shared object file: No such file or dire
ctory
----- End Fatal Exception -------------------------------------------------
```
But I didn't merge #31780 so I guess that's expected

**Comment by mmusich** [comment]
> But I didn't merge #31780 so I guess that's expected

yes...

**Comment by christopheralanwest** [comment]
please test

**Comment by vargasa** [comment]
> I submitted a fix for UnitTestConcurrentIOVsCondCore, #31806

@christopheralanwest #31806 was just merged (thanks @wddgit) that would fix the last unit test

**Comment by christopheralanwest** [comment]
please test

to re-run the tests with in an IB with PR #31806 included

**Comment by christopheralanwest** [comment]
Should the "Draft" status of this PR be removed?

**Comment by ianna** [comment]
+1

**Comment by silviodonato** [comment]
urgent
I have merged #31780 too early

**Comment by kpedro88** [comment]
+upgrade

**Comment by christopheralanwest** [comment]
+alca

**Comment by silviodonato** [comment]
Do you have any comments? @ggovi

**Comment by silviodonato** [comment]
merge
@ggovi I have to merge this PR to fix a failure in a unit test which makes fail all PR tests.
Please let us know if you have any comments/objections on this PR

**Comment by silviodonato** [comment]
We are getting several errors after having merged this PR 
see https://github.com/cms-sw/cmssw/issues/31869

**Pull request merged**
